eclipse/src/saros/filesystem/EclipseReferencePointManager.java

```diff
@@ -0,0 +1,248 @@
+
+    return getFile(referencePoint, eclipsePath);
+  }
+
+  /**
+   * Returns a handle to the {@link IFolder} folder identified by the given {@link IPath} path and
+   * {@link IReferencePoint} reference point
+   *
+   * @param referencePoint The {@link IReferencePoint} reference point, on which the {@link IFolder}
+   *     folder belongs to
+   * @param referencePointRelativePath the {@link IPath} relative path from the {@link
+   *     IReferencePoint} reference point to the {@link IFolder} folder
+   * @return a handle to the {@link IFolder} folder
+   * @exception IllegalArgumentException if {@link IReferencePoint} reference point or {@link IPath}
+   *     referencePointRelativePath is null
+   * @exception NullPointerException if the given {@link IReferencePoint} reference point does't
+   *     exist an {@link IProject} project
+   */
+  public IFolder getFolder(IReferencePoint referencePoint, IPath referencePointRelativePath) {
+    checkArguments(referencePoint, referencePointRelativePath);
+
+    IProject project = Objects.requireNonNull(getProject(referencePoint));
+
+    return project.getFolder(referencePointRelativePath);
+  }
+
+  /**
+   * Returns a handle to the {@link IFolder} folder represented by this {@link SPath} SPath
+   *
+   * @param path the {@link SPath} SPath which represents the {@link IFolder} folder
+   * @return a handle to the {@link IFolder} folder
+   * @exception IllegalArgumentException if {@link SPath} path is null
+   * @exception NullPointerException if the given {@link SPath} path does contain an {@link
+   *     IReferencePoint} reference point
+   * @exception NullPointerException if the given {@link SPath} path does contain an {@link IPath}
+   *     referencePointRelativePath
+   */
+  public IFolder getFolder(SPath path) {
+    checkArgument(path);
+
+    IReferencePoint referencePoint = Objects.requireNonNull(path.getReferencePoint());
+    saros.filesystem.IPath referencePointRelativePath =
+        Objects.requireNonNull(path.getProjectRelativePath());
+
+    IPath eclipsePath = ((EclipsePathImpl) referencePointRelativePath).getDelegate();
+
+    return getFolder(referencePoint, eclipsePath);
+  }
+
+  /**
+   * Check, if the given argument is null
+   *
+   * @param argument which is checked for null
+   * @param <T> Type of argument parameter (like {@link IReferencePoint reference point})
+   * @exception IllegalArgumentException if given argument is null
+   */
+  private <T> void checkArgument(T argument) {
+    if (argument == null) throw new IllegalArgumentException("Argument is null");
+  }
+
+  /**
+   * Check, if the given arguments are null
+   *
+   * @param firstArgument which is checked for null
+   * @param secondArgument which is checked for null
+   * @param <S> Type of first argument
+   * @param <T> Type of second argument
+   * @exception IllegalArgumentException if given arguments are null
+   */
+  private <S, T> void checkArguments(S firstArgument, T secondArgument) {
```

Comment:
> Do I understand the design: The EclipseReferencePointManager has the same purpose as a class that implements the IReferencePointManager, but because of the different return values (classes) the interface is not implemented?

Not only because of the return values. The differences are:

1. The IReferencePointManager maps an IReferencePoint to a Saros project `(saros.filesystem.IProject)`. The EclipseReferencePointManager maps an IReferencePoint to an Eclipse project `(org.eclipse.core.resources.IProject)`. 
2. The IReferencePointManager handles and returns Saros resources `(saros.filesystem.IProject)` and the EclipseReferencePointManager handles and returns Eclipse resources `(org.eclipse.core.resources.*)`, like you wrote.

With the next PR, you will see, that the design of the EclipseReferencePointManager useful is (especially handling with SPaths in Saros/E). 

The design of IntelliJReferencePointManager is, that it handles with IntelliJ Resources (Mapping of IReferencePoints to Modules and returning VirtualFiles in combination of IReferencePoints and IPaths) in a later commit

Reply:
> Is this method really necessary? I am always a fan of implementing small methods, but I see no advantage compared to calling the checkArgument method twice.

No, there no a huge advantage. Your suggestion about `checkArguments(T argument, String message)` is a good idea, so I removed the method, which gets two arguments to check

Reply:
> With the next PR, you will see, that the design of the EclipseReferencePointManager useful is (especially handling with SPaths in Saros/E). I can create a PR, if you want.

Sure, please open the PR. Will the naming clash (EclipseReferencePointManager does not implement IReferencePointManager) be resolved in the following PRs? Otherwise I would propose to change the naming.

Reply:
@pdler Do you still work on this PR?

Reply:
@m273d15 No. I don't know, if I can merge this PR to the `BasedOnReferencePoints` branch.
I will push the next PR in the evening today